postgres_composite_types/__init__.py

```diff
@@ -132,6 +133,36 @@ def formfield(self, **kwargs):  # pylint:disable=arguments-differ
 
         return super().formfield(**defaults)
 
+    def to_python(self, value):
+        """
+        Convert a value to the correct type for this field. Values from the
+        database will already be of the correct type, due to the the caster
+        registered with psycopg2. The field can also be serialized as a string
+        via value_to_string, where it is encoded as a JSON object.
+        """
+        # Composite types are serialized as JSON blobs. If BaseField.to_python
+        # is called with a string, assume it was produced by value_to_string
+        # and decode it
+        if isinstance(value, str):
+            value = json.loads(value)
```

Comment:
I think we need to handle the possibility of this choking, since who knows what random thing we got pointed at. If we can't parse it let's let our super have a go.

Reply:
Good catch, but looking at how the rest of Django handles this case, raising a ValidationError on bad data in `to_python` is the accepted solution. I'll go do that.